system_tests/nox.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Noxfile for automating system tests.
+
+This file handles setting up environments needed by the system tests. This
+separates the tests from their environment configuration.
+
+See the `nox docs`_ for details on how this file works:
+
+.. _nox docs: http://nox.readthedocs.io/en/latest/
+"""
+
+import os
+
+from nox.command import Command
+import py.path
+
+
+HERE = os.path.dirname(__file__)
+DATA_DIR = os.path.join(HERE, 'data')
+SERVICE_ACCOUNT_FILE = os.path.join(DATA_DIR, 'service_account.json')
+AUTHORIZED_USER_FILE = os.path.join(DATA_DIR, 'authorized_user.json')
+CLOUD_SDK_CONFIG_ENV = 'CLOUDSDK_CONFIG'
+
+# If set, this is where the environment setup will store the Cloud SDK.
+# If unset, it will download the SDK to a temporary directory.
+CLOUD_SDK_ROOT = os.environ.get('CLOUD_SDK_ROOT')
+if CLOUD_SDK_ROOT:
+    CLOUD_SDK_ROOT = py.path.local(CLOUD_SDK_ROOT)
+    CLOUD_SDK_ROOT.ensure(dir=True)
+if not CLOUD_SDK_ROOT:
+    CLOUD_SDK_ROOT = py.path.local.mkdtemp()
+
+
+# Helper functions
+
+
+def prerun(*args, **kwargs):
+    """Runs a command before the session."""
+    kwargs.setdefault('silent', True)
+    env = os.environ.copy()
+    env.update(kwargs.pop('env', {}))
+    Command(args, env=env, **kwargs).run()
+
+
+# Cloud SDK helpers
+
+
+def setup_cloud_sdk():
+    """Downloads and installs the Google Cloud SDK."""
+
+    # If the sdk already exists, we don't need to do anything else.
+    if CLOUD_SDK_ROOT.join('google-cloud-sdk').exists():
+        return
```

Comment:
Make a note that you intentionally don't check that it's updated, only that it exists (don't you want to check that it is a directory?)